tool/metrics_monitor.py

```diff
@@ -0,0 +1,75 @@
+import time
+import datetime as dt
+import threading
+import sys
+import psutil
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+from core_data_modules.logging import Logger
+
+COLLECTION = 'pipeline_metrics'
+DEFAULT_INTERVAL = 3
+
+log = Logger(__name__)
+
+
+def initialize_firebase(CRYPTO_TOKEN_PATH):
+    global firebase_client
+    log.info("Setting up Firebase client")
+    firebase_cred = credentials.Certificate(CRYPTO_TOKEN_PATH)
+    firebase_admin.initialize_app(firebase_cred)
+    firebase_client = firestore.client()
+    log.info("Firebase client ready")
+
+
+def get_metrics(interval):
+    while True:
+        metrics = {}
+
+        # record datetime
+        metrics['datetime'] = dt.datetime.now().isoformat() + 'Z'
+
+        # current cpu utlization
+        cpu_utilization = psutil.cpu_percent(interval=0.1)
+        metrics['cpu_percent'] = cpu_utilization
+
+        # cpu load over the last 1, 5 and 15 minutes in percentage
```

Comment:
Ah, right, the list of three values from the sample data makes sense now. Which leads me to wonder if we could be more transparent if we generated a map for this metric instead of a list, e.g. 
```{"1min": <value>, "5mins": <value>, "15mins": <value>}```
That way the view doesn't need to be implicitly aware of what each item in the list is.

Reply:
This is a good idea. I've refactored the changes in